pandas/core/config_init.py

```diff
@@ -279,7 +280,8 @@ def mpl_style_cb(key):
 
 with cf.config_prefix('display'):
     cf.register_option('precision', 6, pc_precision_doc, validator=is_int)
-    cf.register_option('float_format', None, float_format_doc)
+    cf.register_option('float_format', None, float_format_doc,
+                       validator=is_callable)
```

Comment:
@jreback: Assuming my `is_callable` validator raises, this code can't run because `None` isn't callable. `register_option` calls the validator on the default. Not sure how to handle that.